crates/core/src/module_registry.rs

```diff
@@ -0,0 +1,93 @@
+//! Module registry is a service that provides the ability to fetch modules from the StarkNet network.
+//! It fetch contract class from the StarkNet network and compile it to the casm.
+
+use anyhow::{bail, Result};
+use cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+use starknet::{
+    core::types::{BlockId, BlockTag, ContractClass},
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider, Url},
+};
+use starknet_crypto::FieldElement;
+use tracing::info;
+
+use crate::conversion::flattened_sierra_to_compiled_class;
+
+pub struct ModuleRegistry {
+    provider: JsonRpcClient<HttpTransport>,
+}
+
+impl ModuleRegistry {
+    pub fn new(url: Url) -> Self {
```

Comment:
lets use URL type in rest of the project also wherever there is URL type constrain